go/vt/sqlparser/sql.y

```diff
@@ -4690,8 +4690,16 @@ alter_table_statement_part:
   }
 | DROP FOREIGN KEY ID
   {
-    $$ = &DDL{Action: AlterStr, ConstraintAction: DropStr, TableSpec: &TableSpec{Constraints:
-        []*ConstraintDefinition{&ConstraintDefinition{Name: string($4), Details: &ForeignKeyDefinition{}}}}}
+    ddl := &DDL{Action: AlterStr, ConstraintAction: DropStr, TableSpec: &TableSpec{}}
+    ddl.TableSpec.AddConstraint(&ConstraintDefinition{Name: string($4), Details: &ForeignKeyDefinition{}})
+    $$ = ddl
+  }
+| RENAME CONSTRAINT ID TO ID
+  {
+    ddl := &DDL{Action: AlterStr, ConstraintAction: RenameStr, TableSpec: &TableSpec{}}
+    ddl.TableSpec.AddConstraint(&ConstraintDefinition{Name: string($3), Details: &ForeignKeyDefinition{}})
+    ddl.TableSpec.AddConstraint(&ConstraintDefinition{Name: string($5), Details: &ForeignKeyDefinition{}})
```

Comment:
Similar to the other comment... do we have to supply the `Details: &ForeignKeyDefinition{}`, or can we just supply the name and let GMS figure out what type it is? Seems like GMS is going to have to validate it anyway, and this syntax seems useful/intuitive to apply to other constraint types eventually, too.

Reply:
It's probably possible to have GMS determine what kind of constraint it is, but right now `gms/sql/planbuilder/ddl.go:convertConstraintDefinition` switches on `Details` to create the right constraint type.